File: CDCasePrint/CDCasePrint/SQLProcess.vb
Imports System.Data.SqlClient

<System.Diagnostics.DebuggerStepThrough()>
Public Class SQLProcess

#Region "パブリック宣言"

	Dim con As SqlConnection
	Dim cmd As SqlCommand
	Dim dr As SqlDataReader = Nothing

#End Region

#Region "インスタンス"

	''' <summary>
	''' コンストラクタ
	''' </summary>
	Public Sub New()

		'SQLServer接続確認
		Initialize()

	End Sub

	''' <summary>
	''' デストラクタ
	''' </summary>
	Public Sub Close()

		If Not cmd Is Nothing Then
			cmd.Dispose()
		End If
		If Not dr Is Nothing Then
			dr.Close()
		End If
		con.Close()
	End Sub

#End Region

#Region "プライベートメソッド"

	''' <summary>
	''' コネクション確立
	''' </summary>
	Private Sub Initialize()

		XmlSettings.LoadFromXmlFile()

		Try
			con = New SqlConnection
			'接続文字列の設定
			con.ConnectionString =
				"Data Source=" & XmlSettings.Instance.DataSource & ";" &
				"Initial Catalog=" & XmlSettings.Instance.InitialCatalog & ";" &
				"Persist Security Info=True;" &
				"User ID=" & XmlSettings.Instance.UserID & ";" &
				"Password=" & XmlSettings.Instance.Password & ";" &
				"Connection Timeout=600;"

			'Windows認証の場合
			'con.ConnectionString = _
			'    "Data Source = " & settings.DataSource & ";" & _
			'    "Initial Catalog = " & settings.InitialCatalog & ";" & _
			'    "Integrated Security = SSPI"

			con.Open()

		Catch ex As Exception

			Call OutputLogFile("発生場所：" & Reflection.MethodBase.GetCurrentMethod.Name & vbNewLine & ex.Message & vbNewLine & ex.StackTrace)
			MessageBox.Show("発生場所：" & Reflection.MethodBase.GetCurrentMethod.Name & vbNewLine & ex.Message, "エラー", MessageBoxButtons.OK, MessageBoxIcon.Error)

		End Try

	End Sub

#End Region

#Region "パブリックメソッド"

	''' <summary>
	''' SELECT文発行用関数(DATATABLE)
	''' </summary>
	''' <param name="strSQL"></param>
	''' <returns></returns>
	Public Function DB_SELECT_DATATABLE(ByVal strSQL As String) As DataTable

		Try
			cmd = New SqlCommand(strSQL, con)
			cmd.CommandTimeout = 600
			Dim da As SqlDataAdapter = New SqlDataAdapter(cmd)
			Dim dt As New DataTable

			da.Fill(dt)

			Return dt

		Catch ex As Exception

			Call OutputLogFile("発生場所：" & Reflection.MethodBase.GetCurrentMethod.Name & vbNewLine & ex.Message & vbNewLine & strSQL)
			MessageBox.Show("発生場所：" & Reflection.MethodBase.GetCurrentMethod.Name & vbNewLine & ex.Message & vbNewLine & strSQL, "エラー", MessageBoxButtons.OK, MessageBoxIcon.Error)

			Return Nothing

		End Try

	End Function

	''' <summary>
	''' SQL文の結果より単一の値を取得
	''' </summary>
	''' <param name="strSQL"></param>
	''' <returns></returns>
	Public Function DB_EXECUTE_SCALAR(ByVal strSQL As String) As Object

		Try

			cmd = New SqlCommand(strSQL, con)
			cmd.CommandTimeout = 600

			Return cmd.ExecuteScalar()

		Catch ex As Exception

			Call OutputLogFile("発生場所：" & Reflection.MethodBase.GetCurrentMethod.Name & vbNewLine & ex.Message & vbNewLine & strSQL)
			MessageBox.Show("発生場所：" & Reflection.MethodBase.GetCurrentMethod.Name & vbNewLine & ex.Message & vbNewLine & strSQL, "エラー", MessageBoxButtons.OK, MessageBoxIcon.Error)

			Return Nothing

		End Try

	End Function

	''' <summary>
	''' INSERT、DELETE、UPDATE文発行用関数
	''' </summary>
	''' <param name="strSQL"></param>
	Public Sub DB_UPDATE(ByVal strSQL As String)

		Dim sqlTran As SqlTransaction   'トランザクション

		sqlTran = con.BeginTransaction()
		cmd = New SqlCommand(strSQL, con)
		cmd.Transaction = sqlTran
		cmd.CommandTimeout = 600

		Try

			cmd.ExecuteNonQuery()
			sqlTran.Commit()    'コミット

		Catch ex As Exception

			sqlTran.Rollback()  'ロールバック
			Call OutputLogFile("発生場所：" & Reflection.MethodBase.GetCurrentMethod.Name & vbNewLine & ex.Message & vbNewLine & strSQL)
			MessageBox.Show("発生場所：" & Reflection.MethodBase.GetCurrentMethod.Name & vbNewLine & ex.Message & vbNewLine & strSQL, "エラー", MessageBoxButtons.OK, MessageBoxIcon.Error)

		End Try

	End Sub

	''' <summary>
	''' SELECT文発行用関数(DATA READER)
	''' </summary>
	''' <param name="strSQL"></param>
	''' <returns></returns>
	Public Function DB_SELECT_READER(ByVal strSQL As String) As SqlDataReader

		Try

			cmd = New SqlCommand(strSQL, con)
			cmd.CommandTimeout = 600
			dr = cmd.ExecuteReader

			Return dr

		Catch ex As Exception

			Call OutputLogFile("発生場所：" & Reflection.MethodBase.GetCurrentMethod.Name & vbNewLine & ex.Message & vbNewLine & strSQL)
			MessageBox.Show("発生場所：" & Reflection.MethodBase.GetCurrentMethod.Name & vbNewLine & ex.Message & vbNewLine & strSQL, "エラー", MessageBoxButtons.OK, MessageBoxIcon.Error)
			Return Nothing

		End Try

	End Function

	''' <summary>
	''' SELECT文発行用関数(DATASET)
	''' </summary>
	''' <param name="strSQL"></param>
	''' <param name="srcTable"></param>
	''' <returns></returns>
	Public Function DB_SELECT_DATASET(ByVal strSQL As String, ByVal srcTable As String) As DataSet

		Try

			cmd = New SqlCommand(strSQL, con)
			cmd.CommandTimeout = 600
			Dim da As SqlDataAdapter = New SqlDataAdapter(cmd)
			Dim ds As DataSet = New DataSet()
			da.Fill(ds, srcTable)

			Return ds

		Catch ex As Exception

			Call OutputLogFile("発生場所：" & Reflection.MethodBase.GetCurrentMethod.Name & vbNewLine & ex.Message & vbNewLine & strSQL)
			MessageBox.Show("発生場所：" & Reflection.MethodBase.GetCurrentMethod.Name & vbNewLine & ex.Message & vbNewLine & strSQL, "エラー", MessageBoxButtons.OK, MessageBoxIcon.Error)

			Return Nothing

		End Try

	End Function

#End Region

End Class

File: ImageCorruptionCheck/ImageCorruptionCheck/Class/SQLProcess.cs
using System;
using System.Data;
using System.Data.SqlClient;

namespace ImageCorruptionCheck
{
    [System.Diagnostics.DebuggerStepThrough]
    public class SQLProcess
    {
        #region パブリック宣言
        private static SqlConnection con;
        private static SqlCommand cmd;
        private static SqlDataReader dr = null;
        #endregion

        #region インスタンス

        /// <summary>
        /// コンストラクタ
        /// </summary>
        public SQLProcess()
        {
            //SQLServer接続確認
            Initialize();
        }

        /// <summary>
        /// デストラクタ
        /// </summary>
        public void Close()
        {
            if (cmd != null)
            {
                cmd.Dispose();
            }
            if (dr != null)
            {
                dr.Close();
            }
            con.Close();
        }
        #endregion

        #region プライベートメソッド

        /// <summary>
        /// コネクション確立
        /// </summary>
        private static void Initialize()
        {
            XmlSettings.LoadFromXmlFile();

            try
            {
                con = new SqlConnection();
                //接続文字列の設定
                con.ConnectionString =
                    "Data Source=" + XmlSettings.Instance.DataSource + ";" +
                    "Initial Catalog=" + XmlSettings.Instance.InitialCatalog + ";" +
                    "Persist Security Info=True;" +
                    "User ID=" + XmlSettings.Instance.UserID + ";" +
                    "Password=" + XmlSettings.Instance.Password + ";" +
                    "Connection Timeout=600;";

                /*
				//Windows認証の場合
				con.ConnectionString =
					"Data Source=" + XmlSettings.Instance.DataSource + ";" +
					"Initial Catalog=" + XmlSettings.Instance.InirialCatalog + ";" +
					"Integrated Security=SSPI";
				*/
                con.Open();
            }
            catch (Exception)
            {
                //エラーメッセージを表示
                throw;
            }
        }
        #endregion

        #region パブリックメソッド

        /// <summary>
        /// SELECT文発行用関数(DATATABLE)
        /// </summary>
        /// <param name="strSQL"></param>
        /// <returns></returns>
        public DataTable DB_SELECT_DATATABLE(string strSQL)
        {
            try
            {
                cmd = new SqlCommand(strSQL, con);
                cmd.CommandTimeout = 600;
                SqlDataAdapter da = new SqlDataAdapter(cmd);
                DataTable dt = new DataTable();

                da.Fill(dt);

                return dt;
            }
            catch (Exception)
            {
                //エラーメッセージを表示
                throw;
            }
        }

        /// <summary>
        /// SQL文の結果より単一の値を取得
        /// </summary>
        /// <param name="strSQL"></param>
        /// <returns></returns>
        public object DB_EXECUTE_SCALAR(string strSQL)
        {
            try
            {
                cmd = new SqlCommand(strSQL, con);
                cmd.CommandTimeout = 600;

                return cmd.ExecuteScalar();
            }
            catch (Exception)
            {
                //エラーメッセージを表示
                throw;
            }
        }

        /// <summary>
        /// INSERT、DELETE、UPDATE文発行用関数
        /// </summary>
        /// <param name="strSQL"></param>
        public void DB_UPDATE(string strSQL)
        {
            SqlTransaction sqlTran = con.BeginTransaction();    //トランザクション
            cmd = new SqlCommand(strSQL, con);
            cmd.Transaction = sqlTran;
            cmd.CommandTimeout = 600;

            try
            {
                cmd.ExecuteNonQuery();
                sqlTran.Commit();   //コミット
            }
            catch (Exception)
            {
                //エラーメッセージを表示
                throw;
            }
        }

        #endregion
    }
}
